src/isomorphic/classic/types/ReactPropTypes.js

```diff
@@ -226,11 +227,16 @@ function createInstanceTypeChecker(expectedClass) {
 
 function createEnumTypeChecker(expectedValues) {
   if (!Array.isArray(expectedValues)) {
-    return createChainableTypeChecker(function() {
-      return new Error(
-        `Invalid argument supplied to oneOf, expected an instance of array.`
-      );
-    });
+    if (__DEV__) {
+      warning(false, 'Invalid argument supplied to oneOf, expected an instance of array.');
+      return emptyFunction.thatReturnsNull;
+    } else {
+      return createChainableTypeChecker(function() {
```

Comment:
For consistency with DEV behavior, let’s return `emptyFunction.thatReturnsNull` here as well. This way we treat bad argument as early DEV warning, and since we can’t really check propTypes, we don’t.

Not that all these semantics really matter because nobody should be calling prop types in prod but it’s best to have them behave the same way until we explicitly disallow it.